src/Common.OData.ApiExplorer/AspNet.OData/ClassProperty.cs

```diff
@@ -24,13 +24,29 @@ internal ClassProperty( PropertyInfo clrProperty, Type propertyType )
             Attributes = AttributesFromProperty( clrProperty );
         }
 
-        internal ClassProperty( IEnumerable<Assembly> assemblies, IEdmOperationParameter parameter )
+        internal ClassProperty( IServiceProvider services, IEnumerable<Assembly> assemblies, IEdmOperationParameter parameter, IModelTypeBuilder typeBuilder )
         {
```

Comment:
Need to verify that `services` and `typeBuilder` are not `null` here.

Reply:
Also, do you mind throwing in one more small tweak here? `GetPropertyType` was original a read-only field, then I changed it to a method, but now we don't need the method at all any more. Can you revert the field declaration back to:

```c#
internal readonly Type Type;
```

and dump the `GetPropertyType` method altogether?